### PR TITLE
Add Circular Progress till the Firebase Data is fetched

### LINKED ISSUE #75: Latency In Profile avatar sync
example-gif
We can either do some backend stuff to cache the user into our system or if not possible we could place a circular progress bar until the state of the user is known(loggedIn/Not)

### PR DESCRIPTION
**Pull request by Abhishek-kumar09**
Closes #75 
Added Circular Bar till the user is confirmed to be logged in / Not.
*See the effect on refreshing the browser again and again.*

### CODE REVIEW

File: src/reducers/accountReducer.js [hunk 1]
```diff
@@ -11,8 +11,8 @@ import {
 } from 'src/actions/accountActions';
 
 const initialState = {
-  user: null,
```
Comment (KeenWarrior) [inline src/reducers/accountReducer.js:14]: why null was not working for you?

File: src/layouts/MainLayout/TopBar/Account.js [hunk 2]
```diff
@@ -121,12 +130,17 @@ function Account() {
...
             onClick={handleLoginOpen}
             style={{
               textTransform: 'capitalize'
             }}
           >
-            <Typography variant="h6">Login</Typography>
+            {user === undefined ? (
+              <CircularProgress size="24px" color="inherit" />
+            ) : (
+              <Typography variant="h6">Login / Sign Up</Typography>
```
Comment (KeenWarrior) [inline src/layouts/MainLayout/TopBar/Account.js:142]: Shift it back to just ```Login```

**Review by KeenWarrior — CHANGES_REQUESTED** [review]

File: src/reducers/accountReducer.js (hunk 1 above)
Comment (Abhishek-kumar09) [reply src/reducers/accountReducer.js:14]: I got it bit late but we have three states:
1. null: The user is not logged In
2. [Object Object] : The user is logged and have its object like uid and more
3. Undefined: The data is yet not fully fetched from the firebase to guess if it is loogedIn/Not

File: src/layouts/MainLayout/TopBar/Account.js (hunk 2 above)
Comment (Abhishek-kumar09) [reply src/layouts/MainLayout/TopBar/Account.js:142]: On it

**Review by KeenWarrior — APPROVED** [review]

**Pull request merged**